### ISSUE TITLE
Composer TransportException of recipes

### ISSUE THREAD

**Issue by Gokujo** [symfony/flex#826]
### Symfony version(s) affected

5.3

### Description

I am working behind a proxy through Nexus 'cause the local network is not allowed to access the internet.
My problem is that symfony/flex tries to access recipe file on github.

```Bash
[Composer\Downloader\TransportException]
  The "https://api.github.com/repos/symfony/recipes-contrib/contents/index.json?ref=flex/main" file could not be downloaded: php_network_getaddresses: getaddrinfo failed: Der angegebene Host ist unbekannt.   
  failed to open stream: php_network_getaddresses: getaddrinfo failed: Der angegebene Host ist unbekannt.
```

### How to reproduce

composer create-project symfony/website-skeleton web

### Possible Solution

I had to change the const DEFAULT_ENDPOINT in `/vendor/symfony/flex/src/Downloader.php` to 
```PHP
    private const DEFAULT_ENDPOINTS = [
        'http://localhost:91/_assets/composer/symfony/index.json',
        'http://localhost:91/_assets/composer/symfony/contrib/index.json',
    ];
```
I had to download the files locally and than link them to localhost

**Comment by spider4216** [comment]
The had similar issue with symfony 4.4.33 with PHP 7.2.35 when `composer update`

` [Composer\Downloader\TransportException]                                                                                             
  The "https://github.com/repos/symfony/recipes-contrib/contents/index.json?ref=flex/main" file could not be downloaded (HTTP/2 404 )`

But in my case my network always redirect (301) me from api github to another url (server's problem)

You can modify endpoints in composer.json to prevent editing flex file in vendor
![image](https://user-images.githubusercontent.com/10693428/140281701-a7862756-21aa-4e8a-8616-02fd712bdb3c.png)

**Comment by nicolas-grekas** [comment]
Don't you use github for downloading the ZIP of each package?

**Comment by Gokujo** [comment]
> Don't you use github for downloading the ZIP of each package?

Nope, I used the official command 
```BASH
composer create-project symfony/website-skeleton my_project_name
```
I only changed the const in vendor dir

**Comment by nicolas-grekas** [comment]
How do you make composer work if it can't access the internet?
How did flex work before v1.17? Was it silently disabled for example? Another behavior?

**Comment by Gokujo** [comment]
> How do you make composer work if it can't access the internet? How did flex work before v1.17? Was it silently disabled for example? Another behavior?

Like I said in the first post I am using the Nexus proxy, for me it's only one work around.

**Comment by nicolas-grekas** [comment]
In https://github.com/symfony/flex/issues/809#issuecomment-926172154, I suggested making URls in index.json relative.
I think that's the way to go for more flexibility.
Closing meanwhile. Please send a PR if you have some time to work on it!

**Comment by nicolas-grekas** [comment]
Relative URLs implemented in #872
